File: BookServiceJava/src/main/java/org/elcata98/mybooks/booksservice/validator/BookValidator.java
package org.elcata98.mybooks.booksservice.validator;

import org.elcata98.mybooks.booksservice.model.Book;
import org.springframework.stereotype.Component;
import org.springframework.util.Assert;

@Component
public final class BookValidator {

    public static final String CREATE_BOOK_ERROR_MSG = "Cannot create Book using providing a bookId";
    public static final String UPDATE_BOOK_ERROR_MSG = "Cannot change bookId";

    public Book validateCreateBook(final Book book) {

        Assert.isNull(book.getId(), CREATE_BOOK_ERROR_MSG);
        return book;
    }

    public Book validateUpdateBook(final String id, final Book book) {

        Assert.isTrue(id.equals(book.getId()), UPDATE_BOOK_ERROR_MSG);
        return book;
    }
}

File: BooksService/src/main/kotlin/org/elcata98/mybooks/booksservice/validator/BookValidator.kt
package org.elcata98.mybooks.booksservice.validator

import org.elcata98.mybooks.booksservice.model.Book
import org.springframework.stereotype.Component
import org.springframework.util.Assert

@Component
class BookValidator {

    fun validateCreateBook(book: Book): Book {

        Assert.isNull(book.bookId, "Cannot create Book using providing a bookId")

        return book
    }

    fun validateUpdateBook(id: String, book: Book): Book {

        Assert.isTrue(id == book.bookId, "Cannot change bookId")

        return book
    }
}